onnxscript/rewriter/pattern.py

```diff
@@ -205,6 +205,7 @@ def __call__(
         domain: str | None = None,
         version: int | None = None,
         outputs: int | list[str | None] = 1,
+        _allow_other_attributes: bool | None = None,
```

Comment:
Not really.  It is more to avoid conflicts with ONNX attribute names. We discussed doing something similar for other names as well, like domain/version/outputs. Any thoughts about this? May be a good time to do it.

Reply:
I see. I don't have a better idea yet but I am wishing for one

Reply:
(Post internal discussion comment): My personal preference and recommendation is to use kwargs with a leading underscore for domain/version/outputs/ etc. This
* Meets the goal of distinguishing ONNX op attributes from builder's custom options,
* Involves the least typing effort for users and least cognitive overhead in reading-or-writing (except for using names like domain/version, which fails the first criterion).